File: Leetcode/cpp/medium/1689.cpp
/*
 * @lc app=leetcode.cn id=1689 lang=cpp
 * @lcpr version=30204
 *
 * [1689] 十-二进制数的最少数目
 */


// @lcpr-template-start
#include <bits/stdc++.h>
using namespace std;
// @lcpr-template-end
// @lc code=start
#include <ranges>

class Solution {
public:
    int minPartitions(string n) {
        return ranges::max(n) - '0';
    }
};
// @lc code=end



/*
// @lcpr case=start
// "32"\n
// @lcpr case=end

// @lcpr case=start
// "82734"\n
// @lcpr case=end

// @lcpr case=start
// "27346209830709182346"\n
// @lcpr case=end

 */

File: Leetcode/rust/medium/1689.rs
/*
 * @lc app=leetcode id=1689 lang=rust
 *
 * [1689] Partitioning Into Minimum Number Of Deci-Binary Numbers
 */

// @lc code=start
use std::cmp::max;

impl Solution {
    pub fn min_partitions(n: String) -> i32 {
        let mut ans: i32 = 0;
        for c in n.chars() {
            ans = max(ans, (c as u8 - b'0') as i32);
        }
        ans
    }
}
// @lc code=end

